### PR TITLE
Add users endpoint

### PR DESCRIPTION
**Pull request by Urkman**
Add the users endpoint to get the current authenticated user

### CODE REVIEW

**Comment by Urkman** [comment]
added the remote boombox the PR

File: Sources/TeslaSwift/Model/Authentication.swift [hunk 1]
```diff
@@ -7,7 +7,6 @@
 //
 
 import Foundation
-import CryptoKit
```
Comment (jonasman) [inline Sources/TeslaSwift/Model/Authentication.swift:10]: Can you revert all changes here as the current Master branch already has the auth working correctly?

File: Sources/TeslaSwift/Model/User.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        profileImageUrl = try? container.decode(String.self, forKey: .profileImageUrl)
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encodeIfPresent(email, forKey: .email)
+        try container.encodeIfPresent(fullName, forKey: .fullName)
+        try container.encodeIfPresent(profileImageUrl, forKey: .profileImageUrl)
+    }
```
Comment (jonasman) [inline Sources/TeslaSwift/Model/User.swift:36]: this is not needed

File: Sources/TeslaSwift/TeslaSwift.swift [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 import Foundation
 import os.log
+import CryptoKit
+import Security
```
Comment (jonasman) [inline Sources/TeslaSwift/TeslaSwift.swift:12]: also here, all these auth changes can be reverted

File: Sources/TeslaSwift/TeslaSwift.swift [hunk 4]
```diff
@@ -663,3 +698,61 @@ public let teslaJSONDecoder: JSONDecoder = {
         })
 	return decoder
 }()
+
+// MARK: - code to create code_verifier and challenge
```
Comment (jonasman) [inline Sources/TeslaSwift/TeslaSwift.swift:702]: all these stuff isnt needed anymore

**Review by jonasman — CHANGES_REQUESTED** [review]

**Pull request closed without merging**